bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Bookie.java

```diff
@@ -586,6 +586,10 @@ public static BookieSocketAddress getBookieAddress(ServerConfiguration conf)
         if (conf.getUseHostNameAsBookieID()) {
             hostAddress = inetAddr.getAddress().getCanonicalHostName();
         }
+
+        if (conf.getAdvertisedAddress() != null && conf.getAdvertisedAddress().trim().length() > 0) {
```

Comment:
Maybe you can rearrage the code in order to skip completely the previous code, like dns queries...